src/routes/getFiles.ts

```diff
@@ -0,0 +1,246 @@
+import { asBoolean, asMap, asNumber, asObject, asOptional } from 'cleaners'
+import Router from 'express-promise-router'
+
+import config from '../../config.json'
+import { dataStore } from '../db'
+import {
+  ApiResponse,
+  asNonEmptyString,
+  asStoreDirectory,
+  asStoreDirectoryDocument,
+  asStoreFile,
+  asStoreFileDocument,
+  asStoreRepoDocument,
+  StoreDirectoryDocument,
+  StoreFile,
+  StoreFileDocument,
+  StoreFileMap,
+  StoreRepoDocument
+} from '../types'
+import {
+  getNameFromPath,
+  getParentPathsOfPath,
+  makeApiClientError
+} from '../utils'
+
+const { maxPageSize } = config
+
+type GetFilesBody = ReturnType<typeof asGetFilesBody>
+const asGetFilesBody = asObject({
+  repoId: asNonEmptyString,
+  ignoreTimestamps: asOptional(asBoolean),
+  paths: asMap(asNumber)
+})
+
+interface GetFilesResponseData {
+  total: number
+  paths: GetFilesResponsePaths
+}
+
+interface GetFilesResponsePaths {
+  [path: string]: GetFilesResponseFile | GetFilesResponseDirectory
+}
+
+interface GetFilesResponseFile extends StoreFile {
+  timestamp: number
+}
+
+interface GetFilesResponseDirectory {
+  paths: StoreFileMap
+  timestamp: number
+}
+
+export const getFilesRouter = Router()
+
+getFilesRouter.post('/getFiles', async (req, res) => {
+  let body: GetFilesBody
+  let repoId: string
+  let ignoreTimestamps: boolean = false
+
+  // Validate request body
+  try {
+    body = asGetFilesBody(req.body)
+    repoId = body.repoId
+
+    if (body.ignoreTimestamps !== undefined) {
+      ignoreTimestamps = body.ignoreTimestamps
+    }
+
+    if (repoId === '') {
+      throw new Error(`Missing repoId.`)
+    }
+  } catch (err) {
+    throw makeApiClientError(400, err.message)
+  }
+
+  const responsePaths: GetFilesResponsePaths = {}
+
+  const bodyPaths = Object.keys(body.paths)
+
+  // Use max page size config to limit the paths processed
+  if (bodyPaths.length > maxPageSize) {
+    throw makeApiClientError(
+      422,
+      `Too many paths. maxPageSize is ${maxPageSize}`
+    )
+  }
+
+  const bodyKeys = bodyPaths.map(filePath => `${repoId}:${filePath}`)
+  // Maps a directory key to a file path for timestamp lookup
+  const filePathMapOfParentDirKey: { [dirKey: string]: string[] } = {}
+
+  // Fetch Paths:
+
+  const fetchPathsResults = await dataStore.fetch({ keys: bodyKeys })
+
+  for (const row of fetchPathsResults.rows) {
+    const documentKey = row.key
+    // No need to check for undefined case here because of how keys
+    // are defined for the fetch.
```

Comment:
Not sure what you mean here. The comment is talking about how we don't need to check for the undefined case for `documentKey.split[1]` because of we know that the key should be correct in couchDB. Is there an issue that needs addressing here?